components/raftstore/src/store/fsm/store.rs

```diff
@@ -614,9 +614,23 @@ impl<'a, EK: KvEngine + 'static, ER: RaftEngine + 'static, T: Transport>
                     self.on_store_unreachable(store_id);
                 }
                 StoreMsg::Start { store } => self.start(store),
-                StoreMsg::CheckLeader { leaders, cb } => self.on_check_leader(leaders, cb),
+                StoreMsg::CheckLeader { leaders, cb } => {
```

Comment:
Why have to go through raftstore?

Reply:
I have tried to redirect the request at the server layer through the `resolved-ts` worker scheduler but failed due to cycle dependency of `resolved-ts` and `tikv` crate.

Reply:
I don't get it. Seems what's needed is just a channel sender, why need to include all the dependency?

Reply:
It needs the worker's `Task`, but since we move the `CheckLeader` handler to another worker, I think we can use that worker's scheduler to redirect requests now.